### DISCUSSION TITLE
space between parens () and brakets []

### DISCUSSION THREAD

**Discussion by Reed-Anderson** [biomejs/biome#982] [Ideas]
Hi folks, just wondering if the Biome team would be more open to the implementation of this Prettier request that was declined:

https://github.com/prettier/prettier/issues/1303

This formatting is available in both ESLint and DPrint, but as far as I can tell, not currently possible with Biome.

Function Declaration:
```
function foo( one, two, three ){
```
Instead of
```
function foo(one, two, three){
```
Arrays:
```
var array = [ 1, 2, 3, 4 ]
```
instead of:
```
var array = [1, 2, 3, 4]
```
Methods:
```
array.slice( 0, 1 )
```
instead of
```
array.slice(0, 1)
```
Array indexing:
```
array[ 0 ]
```
instead of 
```
array[0]
```

This issue was a Prettier deal breaker for some that I work with. Biome's support would definitely win them over.

**Comment by anatooly** [comment]
How return spaces?

**Comment by antony** [comment]
This is a dealbreaker for us - array spacing is very readable when grokking code.

**Comment by dhasilva** [comment]
There is an issue [here](https://github.com/biomejs/biome/issues/4607) with a draft PR by @sirreal. Apparently @ematipico is not opposed to the idea, as suggested on his comment [here](https://github.com/biomejs/biome/pull/4605#issuecomment-2488926797), so there is hope.
